### PR TITLE
systemd: provide more detailed errors for unimplemented method in emulation mode

### PR DESCRIPTION
**Pull request by stolowski**
Provide more details with `notImplementedError` error to make it easier to understand what goes wrong.

### CODE REVIEW

File: systemd/emulation.go [hunk 1]
```diff
@@ -39,18 +38,24 @@ type emulation struct {
 	rootDir string
 }
 
-var errNotImplemented = errors.New("not implemented in emulation mode")
+type errNotImplemented struct {
```
Comment (mvo5) [inline systemd/emulation.go:41]: (nitpick) I would now call it `notImplementedError`, the pattern in go seems to be that the `ErrXXXX` prefix is used for simple errors like `ErrWriteToConnected = errors.New(...)` and the `xxxxError` postfix when there are error structs. I vaguely remember that this was also written down somewhere but I can't find it anymore.

**Review by mvo5 — COMMENTED** [review]
Thanks, this looks very useful. One naming bikeshed/nitpick.

File: systemd/emulation.go (hunk 1 above)
Comment (stolowski) [reply systemd/emulation.go]: Done.

File: systemd/emulation.go [hunk 2]
```diff
@@ -64,59 +69,59 @@ func (s *emulation) Disable(service string) error {
...
 }
 
 func (s *emulation) ReloadOrRestart(service string) error {
-	return errNotImplemented
+	return &notImplementedError{"ReloadOrRestart"}
 }
 
 func (s *emulation) RestartAll(service string) error {
-	return errNotImplemented
+	return &notImplementedError{"RestartAlll"}
```
Comment (MiguelPires) [inline systemd/emulation.go:96]:
```suggestion
	return &notImplementedError{"RestartAll"}
```

**Review by MiguelPires — APPROVED** [review]
LGTM, there's just one tiny typo

File: systemd/emulation.go (hunk 2 above)
Comment (stolowski) [reply systemd/emulation.go]: Thanks!

**Review by mardy — APPROVED** [review]
LGTM!

**Review by mvo5 — APPROVED** [review]
\o/

**Pull request merged**